expr/plan.go

```diff
@@ -238,124 +335,49 @@ func newplanFunc(e *expr, fn Func, from, to uint32, stable bool, reqs []Req) ([]
 	// for any provided keyword args, verify that they are what the function stipulated
 	// and that they have not already been specified via their position
 	for k := range e.namedArgs {
-		err = consumeKwarg(argsOptExp, e.namedArgs, k, seenKwargs)
+		err = consumeKwarg(e.namedArgs, k, argsExp[cutoff:], seenKwargs)
 		if err != nil {
 			return nil, err
 		}
 
 	}
-	err = fn.Init(e.args, e.namedArgs)
-	if err != nil {
-		return nil, err
-	}
+
+	// functions now know all non-series input args -> they know the timerange they need
 	from, to = fn.NeedRange(from, to)
-	// look at which arguments are requested
-	// if the args are series, they are to be requested with the potentially extended to/from
-	// if they are not, keep traversing the tree until we find out which metrics to fetch and for which durations
-	for _, arg := range e.args {
-		if arg.etype == etName || arg.etype == etFunc {
-			reqs, err = newplan(arg, from, to, stable, reqs)
+	// we know timerange -> set up input series
```

Comment:
this comment adds confusion not clarity. I have no idea what it is trying to say.